runtime/sema/check_composite_declaration.go

```diff
@@ -1938,29 +2097,42 @@ func (checker *Checker) checkCompositeFunctions(
 	}
 }
 
-func (checker *Checker) declareSelfValue(selfType Type, selfDocString string) {
-
-	// NOTE: declare `self` one depth lower ("inside" function),
+func (checker *Checker) declareLowerScopedValue(
+	ty Type,
+	docString string,
+	identifier string,
+	kind common.DeclarationKind,
+) {
+	// NOTE: declare value one depth lower ("inside" function),
 	// so it can't be re-declared by the function's parameters
 
 	depth := checker.valueActivations.Depth() + 1
 
-	self := &Variable{
-		Identifier:      SelfIdentifier,
+	variable := &Variable{
+		Identifier:      identifier,
 		Access:          ast.AccessPublic,
-		DeclarationKind: common.DeclarationKindSelf,
-		Type:            selfType,
+		DeclarationKind: kind,
+		Type:            ty,
 		IsConstant:      true,
 		ActivationDepth: depth,
 		Pos:             nil,
-		DocString:       selfDocString,
+		DocString:       docString,
 	}
-	checker.valueActivations.Set(SelfIdentifier, self)
+	checker.valueActivations.Set(identifier, variable)
 	if checker.PositionInfo != nil {
-		checker.recordVariableDeclarationOccurrence(SelfIdentifier, self)
+		checker.recordVariableDeclarationOccurrence(identifier, variable)
 	}
 }
 
+func (checker *Checker) declareSelfValue(selfType Type, selfDocString string) {
+	checker.declareLowerScopedValue(selfType, selfDocString, SelfIdentifier, common.DeclarationKindSelf)
+}
+
+func (checker *Checker) declareSuperValue(baseType Type, superDocString string) {
+	superType := NewReferenceType(checker.memoryGauge, baseType, false)
```

Comment:
References to `super` should be non-auth, as the actual base type in practice may be any number of subtypes of the annotated base type, not all of which should be available to the writer of the attachment.

Reply:
Great comment, please add it to the code